### PR TITLE
Remove mapbox access-token from debug pages and use custom style environment

### PR DESCRIPTION
**Pull request by Joxit**
The folder `debug` is useful for debugging and see if something goes wrong.
Since MapLibre will cut all links with mapbox, we should be able to test our features with some simple maps...

In this PR I added a new index.html file with a new environment variable `MAPLIBRE_MAP_STYLE` to set your custom style for debug pages.

## How To
It's pretty simple, you can easily run the debug server via npm run
```sh
# Use a custom style, this will be the default style on all pages, you can add this line in your .bashrc
export MAPLIBRE_MAP_STYLE="https://api.jawg.io/styles/jawg-streets.json?access-token=..."
# The server will start on port 9966, go to http://localhost:9966/debug you will see a list of all debug pages
npm run start-debug
```

New `index.hyml`
image

`heatmap.html`
image

There are still pages that will not be supported, such as the pages with raster DEM or satellite which are not available in all providers.

### CODE REVIEW

File: build/generate-map-style-script.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/* eslint-disable */
+'use strict';
+const fs = require('fs');
+const path = require('path');
+const script = fs.readFileSync(path.join(__dirname, '../debug/map_style.js'), 'utf-8')
+    .replace('process.env.MapLibreMapStyle',
```
Comment (nyurik) [inline build/generate-map-style-script.js:6]: why introduce two new env vars? Most env vars are caps anyway, might as well use just one

**Review by nyurik — COMMENTED** [review]
I agree that we should remove the token generation. I do not think we need to make the style thing dynamic though -- we might as well just hardcode them, e.g. take them from maptiler.com or have some simple local service that starts with tests.

**Comment by HarelM** [comment]
I've just tried to run the debug page and it failed (the production failed to, probably on token generation), and found out that there's a pull request to basically fix this.
I think that a default style should be used for testing instead of introducing an environment variable and generated code which will make things harder for someone who just wishes to build and run the project to see that everything is working as expected.
Much like the documentation pages, we have a style that was donated (I think) and tiles that were donated which we can use.
If someone would like to test his style and feature he can always change the html files manually and test it without committing.
Also there some code review changes which require fixing. @Joxit let me know if you need my help in order to push this forward.
Also when running the debug page it would be great if there was a console message saying something like: "now listening at 0.0.0.0:9966" to notify the developer a server is up and listening...

**Comment by HarelM** [comment]
@Joxit any update regarding my comment above?
I would really love to get this in for easier debug and development experience, but I think a few changes are still needed...

**Comment by Joxit** [comment]
Hi there, this PR is outdated, we should use `https://demotiles.maplibre.org/style.json` instead of supporting custom map style.

I think we will not be able to keep all the debug pages because they are not all compatible with demotiles.

**Comment by HarelM** [comment]
I agree, but let's do an incremental update to use the demotiles in all examples and later on we can fix the examples that are not working.

**Comment by Joxit** [comment]
Update, I removed all styles from mapbox in favor of demotiles.

I can't update the server wording `now listening at 0.0.0.0:9966` because they use `run-p` and `st` instead of `rollup-plugin-serve`. I prefer to do this change in another PR

**Review by HarelM — APPROVED** [review]
Cool! thanks!

**Comment by HarelM** [comment]
@joxit how do I see the above index.hyml page you mentioned? I couldn't find a way to get to it...
When running `npm start-debug` and surf to `localhost:9966/debug` only shows a simple map and not an index page... is this expected?

**Comment by Joxit** [comment]
Yes, I was not satisfied with it, so I deleted it.

**Comment by HarelM** [comment]
Oh, I see. I guess we'll need to write some documentation about how to test changes your changes in the local development environment and what to run and try out...
I currently do not know, so I'll be happy to understand... :-) 
unless you think the contribution guide is good, because everything I tried failed, but it also might be related to the node version I have which is addressed here: #190

**Comment by xabbu42** [comment]
I do not think the demotiles and style are suitable for easy and quick debuging/testing as they are too simple. Its nice to have something that just works out of the box, but it should still be as easy as possible to test changes with bigger style covering more of the maplibre-gl-js features. for example I used the debug page to quickly test #145 and this is no longer possible. Should I open an issue for this?

**Comment by HarelM** [comment]
I would simply open a PR. I think we a agree that's needed in case you have a solution in mind.

**Comment by xabbu42** [comment]
Not everything can be solved by a lone wolf pull requests. I do not know what exactly you mean by "that" and I can see different possible solutions. Also it seems @Joxit has more pull requests planed in that area and I'd rather not duplicate work without coordinating. So do we have an appropriate place to **discuss** this or are there already plans to improve the situation?

**Comment by HarelM** [comment]
As this is fully community driven, I'm under the impression most contribution will be "lone wolf" style. I might be wrong.
Feel free to open an issue, discission, PR. Whatever works for you :-)

**Comment by Joxit** [comment]
IMO, you can add a new debug page for #145 using demotiles with a new custom layer and your data.

Maintaining all debug examples is too painful, many of them will not work because they are based on mapbox layer names or use specific sources (satellite, raster DEM...). So even if we can add our own tile server, there is no guaranties.

That's why I wanted to keep it as simple as possible.

**Comment by HarelM** [comment]
@Joxit fair enough, I get your point.
Let's just make sure we document it somewhere - I.e. How to test your code/feature/changes.

Bare in mind that we are maintaining examples in the docs repo. We might need to think of an easy way to maintain both.

This is what was done in ngx-maplibre-gl by testing the docs as part of the build. I think it's a good solution.

It does requests require some work to bring it up.

In any case, I'm good with whatever works for you.

**Comment by xabbu42** [comment]
@Joxit I know that I can adapt the debug page and thats what I will do going forward with #145, which was just an example. But for new potential developers it would be better if they have an easier way to just make a quick change and test a reasonably complex map with their changes. I also mostly care about debug/index.html, not about all the other files in debug/. I opened an issue to discuss possible solutions and my preferred solution would not add any complexity if it is possible.

**Pull request merged**